packages/next/server/config-shared.ts

```diff
@@ -9,8 +9,13 @@ export type DomainLocales = Array<{
   defaultLocale: string
 }>
 
+type NoOptionals<T> = {
+  [P in keyof T]-?: T[P]
+}
+
+export type NextConfigComplete = NoOptionals<NextConfig>
```

Comment:
Won't this remove all properties since every property in next.config.js is optional?

Reply:
It should only be converting the optional fields to non-optional to prevent us from having to add `config.someField!` every time we access it internally even though we already check the values during loading